arm/mysql/models.go

```diff
@@ -232,31 +316,108 @@ type ServerProperties struct {
 	FullyQualifiedDomainName *string            `json:"fullyQualifiedDomainName,omitempty"`
 }
 
+// IServerPropertiesForCreate is interface used for polymorphic Properties in ServerForCreate
+type IServerPropertiesForCreate interface {
```

Comment:
The 'I' prefix is a C#-ism, I think we should drop it and simply call the type ServerPropertiesForCreate.